pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -226,6 +226,12 @@ stdenv.mkDerivation (rec {
       stripLen = 3;
       extraPrefix = "libraries/Cabal/Cabal/";
     })
+
+    # We need to be able to set AR_STAGE0 and LD_STAGE0 when cross-compiling
+    (fetchpatch {
+      url = "https://gitlab.haskell.org/ghc/ghc/-/commit/8f7dd5710b80906ea7a3e15b7bb56a883a49fed8.patch";
+      hash = "sha256-C636Nq2U8YOG/av7XQmG3L1rU0bmC9/7m7Hty5pm5+s=";
+    })
```

Comment:
> because that patch patches `configure`, not `configure.ac`, and `8.10.7` uses `autoreconfHook` (but `8.8.4` doesn't).

Note that the copy-paste-tweak style that our GHC expression uses makes these kinds of footguns inevitable.  You have to stare at the code for a long time to realize when somebody did something six times but the seventh time they did it differently.

@sternenseemann you guys really should consider doing something like this to the GHC expression:

- https://github.com/NixOS/nixpkgs/pull/249707

...otherwise it's way too easy for bugs to slip in during all-versions refactorings like this PR.

Reply:
I'd have used `autoreconfHook` everywhere, but it unfortunately didn't work for 8.8.4. Patching `configure` was the simplest fix I could come up with.

Reply:
By the way, the configure patch is redundant now that 8.8 has been removed, so I have removed it altogether.